alpha/apps/kaltura/lib/clipconcat/kClipManager.php

```diff
@@ -0,0 +1,509 @@
+	{
+		KalturaLog::info('Going To Start Concat Job');
+		if($batchJob->getJobType() != BatchJobType::CLIP_CONCAT)
+		{
+			return;
+		}
+
+		foreach ($batchJob->getChildJobs() as $job)
+		{
+			/** @var BatchJob $job */
+			KalturaLog::info('Child job id [' . $job->getId() . '] status [' . $job->getStatus() . ']' . '] type ['.$job->getJobType() .']' );
+			KalturaLog::info('Flavor Param Ids:' .$job->getEntry()->getFlavorParamsIds());
+		}
+		/** @var kClipConcatJobData $jobData */
+		$jobData = $batchJob->getData();
+
+		$tempEntry = entryPeer::retrieveByPK($jobData->getTempEntryId());
+		$assets = assetPeer::retrieveByEntryId($jobData->getTempEntryId());
+		usort($assets, array("kClipManager","cmpByOrder"));
+
+		$files = $this->getFilesPath($assets);
+
+		$flavorAsset = $this->getNewAssetFromEntry($tempEntry);
+
+		kJobsManager::addConcatJob($batchJob, $flavorAsset, $files,false);
+
+	}
+
+	/**
+	 * @param $a flavorAsset
+	 * @param $b flavorAsset
+	 * @return int
+	 */
+	private function cmpByOrder($a,$b)
+	{
+		$aClipNumber = $a->getFromCustomData(self::CLIP_NUMBER);
+		$bClipNumber = $b->getFromCustomData(self::CLIP_NUMBER);
+		if (!$aClipNumber )
+		{
+			return -1;
+		}
+		if (!$bClipNumber )
+		{
+			return 1;
+		}
+		return  ($aClipNumber < $bClipNumber) ? -1 : 1;
+
+	}
+
+
+	/***
+	 * @param BatchJob $batchJob
+	 * @throws APIException
+	 * @throws PropelException
+	 */
+	private function handleClipConcatParentJob($batchJob)
+	{
+		switch ($batchJob->getStatus()) {
+			case BatchJob::BATCHJOB_STATUS_PENDING:
+				$errDesc = '';
+				/**@var kClipConcatJobData $jobData */
+				$jobData = $batchJob->getData();
+				$this->addClipJobs($batchJob, $jobData->getTempEntryId(), $errDesc,
+					$jobData->getPartnerId(),
+					$jobData->getOperationAttributes(), $jobData->getPriority());
+				kJobsManager::updateBatchJob($batchJob, BatchJob::BATCHJOB_STATUS_PROCESSING);
+				break;
+
+			default:
+				break;
+		}
+	}
+
+	/**
+	 * @param BatchJob $parentJob  clipConcat job
+	 * @param $entryId
+	 * @param $errDescription
+	 * @param $partnerId
+	 * @param array $operationAttributes
+	 * @param int $priority
+	 * @return BatchJob[]
+	 * @throws APIException
+	 * @throws PropelException
+	 */
+	private function addClipJobs($parentJob , $entryId, &$errDescription, $partnerId,
+	                             array $operationAttributes, $priority = 0)
+	{
+		$batchArray = array();
+		$order = 0;
+		foreach($operationAttributes as $singleAttribute)
+		{
+			KalturaLog::info("Going To create Flavor for clip: " . print_r($singleAttribute));
+			$clonedID =	$this->cloneFlavorParam($singleAttribute->getAssetParamsId());
+			$flavorAsst = $this->createTempClipFlavorAsset($partnerId,$entryId,$clonedID,$order);
+			$batchJob =	kBusinessPreConvertDL::decideAddEntryFlavor($parentJob, $entryId,
+					$clonedID, $errDescription,$flavorAsst->getId()
+					, array($singleAttribute) , $priority);
+			if(!$batchJob) //todo error;
```

Comment:
Need todo something here , error to log , or maybe raise exception that will break the flow.